Python/optimizer.c

```diff
@@ -394,6 +394,29 @@ executor_traverse(PyObject *o, visitproc visit, void *arg)
     return 0;
 }
 
+static PyObject *
+get_jit_code(PyObject *self, PyObject *Py_UNUSED(ignored))
+{
+    if (!Py_IS_TYPE(self, &_PyUOpExecutor_Type)) {
+        PyErr_SetString(PyExc_TypeError, "get_jit_code() requires a uop_executor object.");
+        return NULL;
+    }
+    _PyExecutorObject *executor = (_PyExecutorObject *)self;
+    if (executor->jit_code == NULL || executor->jit_size == 0) {
+        PyErr_SetString(PyExc_ValueError, "No JIT code available.");
+        return NULL;
+    }
+    return PyBytes_FromStringAndSize(executor->jit_code, executor->jit_size);
```

Comment:
When `_Py_JIT` is not defined, the code may land here (since `jit_code` and `jit_size` are apparently uninitialized).

Reply:
hm, yes it looks like jit_code is only set to NULL when _Py_JIT is defined. To avoid any undefined behaviour, I've added a fixed exception when the runtime was compiled without the JIT in https://github.com/python/cpython/pull/117959/commits/7be0d85cc0f6d3fcc87f854bbd979d02b55a443d